webserver/jersey/src/main/java/io/helidon/webserver/jersey/ResponseWriter.java

```diff
@@ -178,18 +139,185 @@ public boolean enableResponseBuffering() {
         return false;
     }
 
-    /**
-     * Flush buffer if using SSE or if an empty buffer is received for writing. See
-     * {@link io.helidon.common.reactive.MultiFromOutputStream#flush()}.
-     * Manual flushing is required to support
-     * {@link javax.ws.rs.core.StreamingOutput} in MP.
-     *
-     * @param context The container response.
-     * @param byteBuffer The byte buffer to write.
-     * @return Outcome of test.
-     */
-    private static boolean doFlush(ContainerResponse context, ByteBuffer byteBuffer) {
-        return MediaType.SERVER_SENT_EVENTS_TYPE.isCompatible(context.getMediaType())
-                || byteBuffer.hasArray() && byteBuffer.array().length == 0;
+    private static class DataChunkOutputStream extends OutputStream
+            implements Flow.Publisher<DataChunk>, Flow.Subscription {
+
+        private static final int BYTEBUF_DEFAULT_SIZE = 4096;
+        private static final long CANCEL = Long.MIN_VALUE;
+        private static final long ERROR = CANCEL + 1;
+        private static final long WAIT = -1;
+        private static final ByteBuf ZERO_BUF = Unpooled.buffer(0);
+
+        private byte[] oneByteArray;
+        private ByteBuf byteBuf;
+        private ByteBuf byteBufRef;
+        private boolean autoFlush;
+        private volatile Flow.Subscriber<? super DataChunk> downstream;
+        private volatile Semaphore sema;
+        private final AtomicLong requested = new AtomicLong();
+
+        public void autoFlush(boolean autoFlush) {
+            this.autoFlush = autoFlush;
+        }
+
+        // -- OutputStream -----------------------------------------------------
+
+        @Override
+        public void write(int b) throws IOException {
+            if (oneByteArray == null) {
+                oneByteArray = new byte[1];
+            }
+            oneByteArray[0] = (byte) b;
+            write(oneByteArray, 0, 1);
+        }
+
+        @Override
+        public void write(byte[] b, int off, int len) throws IOException {
+            while (len > 0) {
+                if (byteBuf == null) {
+                    awaitRequest();
+                    byteBuf = PooledByteBufAllocator.DEFAULT.buffer(BYTEBUF_DEFAULT_SIZE);
```

Comment:
@spericas This was what I was asking about earlier today.  This will not make use of direct buffers, and perhaps that is just fine.  On the other hand, if you use `ByteBufAllocator.DEFAULT` you'll get [some perhaps useful behavior](https://github.com/netty/netty/blob/1529ef1794e0a6654fa4334fd979b769d6940e61/buffer/src/main/java/io/netty/buffer/ByteBufUtil.java#L77-L93).

Reply:
Sorry; I got slightly confused here.  `PooledByteBufAllocator` will indeed use direct buffers.  I guess my comment is more: if you simply use `ByteBufAllocator.DEFAULT` you pick up whatever default behavior Netty wants in the future.

Reply:
But If we use `ByteBufAllocator` today, are we guaranteed to get a _pooled_ buffer now? After all, that is the whole point of this PR.

Reply:
If you see https://github.com/netty/netty/blob/1529ef1794e0a6654fa4334fd979b769d6940e61/buffer/src/main/java/io/netty/buffer/ByteBufUtil.java#L77-L93 you can see what it does.  In terms of guarantees, there is very little documentation so effectively no contract as to what you can expect or depend upon.